### PR TITLE
Major release 3.0.0

### PR DESCRIPTION
**Pull request by Jeasmine**
Add Android major release support

```
* Support for DeviceState
* Remove focus display functions
* Add OS prefix to interfaces
* Remove PermissionSubscriptionState
* Change from init to setAppId
```

```
Add iOS major release support

* Support for DeviceState
* Remove focus display functions
* Remove PermissionSubscriptionState
* Change from init to setAppId
```

---
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-flutter-sdk/313)

### CODE REVIEW

**Comment by tanmoy27112000** [comment]
This issue is creating a problem with the existing project on flutter and other native platforms.

**Comment by jmadler** [comment]
> This issue is creating a problem with the existing project on flutter and other native platforms.

Hi tanmoy27112000, I'm not sure what issue you're describing.  This PR is to port in the new SDK major release into the Flutter SDK.  Can you please file a new issue if you have any concerns?  Thanks!

**Review by jkasten2 — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/onesignal/flutter/OneSignalSerializer.java [hunk 1]
```diff
@@ -41,21 +55,17 @@
...
-            hash.put("subscriptionStatus", convertSubscriptionStateToMap(subState));
-
-        if (permState != null)
-            hash.put("permissionStatus", convertPermissionStateToMap(state.getPermissionStatus()));
-
-        if (emailState != null)
-            hash.put("emailSubscriptionStatus", convertEmailSubscriptionStateToMap(state.getEmailSubscriptionStatus()));
+        hash.put("areNotificationsEnabled", state.areNotificationsEnabled());
+        hash.put("pushDisabled", state.isPushDisabled());
+        hash.put("subscribed", state.isSubscribed());
```
[the remark this answers is not in the document]
Comment (Jeasmine) [reply android/src/main/java/com/onesignal/flutter/OneSignalSerializer.java:64]: nice catch! my bad 😞  still missing the is prefix

**Comment by mingosnunes** [comment]
Only tested on Android. I just updated the package on my existing project.

I want to test this version but the app terminates when it receives a new push notification in foreground. When the app is closed, the notification pop-up appears but the setNotificationOpenedHandler is not triggered.

**Foreground Error:**
> OneSignal(24514): OneSignalRestClient: After con.getResponseCode to: https://api.onesignal.com/players/5a56a8a3-bf4b-4ed3-aec9-bec223662f4c/on_session
D/OneSignal(24514): OneSignalRestClient: Failed request to: https://api.onesignal.com/players/5a56a8a3-bf4b-4ed3-aec9-bec223662f4c/on_session
W/OneSignal(24514): OneSignalRestClient: POST RECEIVED JSON: {"errors":["[\"Identifier invalid format.\"]"]}
W/OneSignal(24514): Failed last request. statusCode: 400
W/OneSignal(24514): response: {"errors":["[\"Identifier invalid format.\"]"]}
V/OneSignal(24514): initWithContext called with: android.app.ReceiverRestrictedContext@52807e1
V/OneSignal(24514): Starting OneSignal initialization!
V/OneSignal(24514): No class found, not setting up OSRemoteNotificationReceivedHandler
D/OneSignal(24514): OneSignal SDK initialization already completed.
D/OneSignal(24514): OSNotificationWorkManager enqueueing notification work with notificationId: aec40e9a-c0c4-4908-b8e6-44f0b3809be7 and jsonPayload: {"google.delivered_priority":"normal","google.sent_time":1611946374791,"google.ttl":259200,"google.original_priority":"normal","custom":"{\"a\":{\"addition data\":\"1234\",\"actionButtons\":[],\"actionId\":\"__DEFAULT__\"},\"i\":\"aec40e9a-c0c4-4908-b8e6-44f0b3809be7\"}","pri":"5","vis":"1","from":"214972727859","alert":"asdasd","title":"Test Slack Alert","google.message_id":"0:1611946374815968%f54ecb25f9fd7ecd","google.c.sender.id":"214972727859"}
D/OneSignal(24514): NotificationWorker running doWork with data: androidx.work.Data@d0579111
D/OneSignal(24514): Running destroyTimeout with runnable: com.onesignal.OSNotificationReceivedEvent$1@5f80f8c
D/OneSignal(24514): Running startTimeout with timeout: 25000 and runnable: com.onesignal.OSNotificationReceivedEvent$1@5f80f8c
W/OneSignal(24514): remoteNotificationReceivedHandler not setup, displaying normal OneSignal notification
D/OneSignal(24514): Running destroyTimeout with runnable: com.onesignal.OSNotificationReceivedEvent$1@5f80f8c
D/OneSignal(24514): Starting processJobForDisplay opened: false fromBackgroundLogic: true
I/OneSignal(24514): Fire notificationWillShowInForegroundHandler
D/OneSignal(24514): Running destroyTimeout with runnable: com.onesignal.OSNotificationReceivedEvent$1@1a376d5
D/OneSignal(24514): Running startTimeout with timeout: 25000 and runnable: com.onesignal.OSNotificationReceivedEvent$1@1a376d5
D/onesignal(24514): Created json raw payload: {vis=1, google.original_priority=normal, alert=asdasd, google.sent_time=1611946374791, pri=5, google.delivered_priority=normal, custom={"a":{"addition data":"1234","actionButtons":[],"actionId":"__DEFAULT__"},"i":"aec40e9a-c0c4-4908-b8e6-44f0b3809be7"}, google.c.sender.id=214972727859, google.ttl=259200, from=214972727859, title=Test Slack Alert, google.message_id=0:1611946374815968%f54ecb25f9fd7ecd}
D/AndroidRuntime(24514): Shutting down VM
E/AndroidRuntime(24514): FATAL EXCEPTION: main
E/AndroidRuntime(24514): Process: com.domingosnunes.scorenoobapp, PID: 24514
E/AndroidRuntime(24514): java.lang.IllegalArgumentException: Unsupported value: {"addition data":"1234"}
E/AndroidRuntime(24514): 	at io.flutter.plugin.common.StandardMessageCodec.writeValue(StandardMessageCodec.java:278)
E/AndroidRuntime(24514): 	at io.flutter.plugin.common.StandardMessageCodec.writeValue(StandardMessageCodec.java:275)
E/AndroidRuntime(24514): 	at io.flutter.plugin.common.StandardMethodCodec.encodeMethodCall(StandardMethodCodec.java:38)
E/AndroidRuntime(24514): 	at io.flutter.plugin.common.MethodChannel.invokeMethod(MethodChannel.java:99)
E/AndroidRuntime(24514): 	at io.flutter.plugin.common.MethodChannel.invokeMethod(MethodChannel.java:83)
E/AndroidRuntime(24514): 	at com.onesignal.flutter.FlutterRegistrarResponder$4.run(FlutterRegistrarResponder.java:73)
E/AndroidRuntime(24514): 	at android.os.Handler.handleCallback(Handler.java:883)
E/AndroidRuntime(24514): 	at android.os.Handler.dispatchMessage(Handler.java:100)
E/AndroidRuntime(24514): 	at android.os.Looper.loop(Looper.java:214)
E/AndroidRuntime(24514): 	at android.app.ActivityThread.main(ActivityThread.java:7682)
E/AndroidRuntime(24514): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(24514): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
E/AndroidRuntime(24514): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
I/WM-WorkerWrapper(24514): Worker result SUCCESS for Work [ id=517e0eef-1a39-4340-9361-03eb804bf6ea, tags={ com.onesignal.OSNotificationWorkManager$NotificationWorker } ]
D/TransportRuntime.SQLiteEventStore(24514): Storing event with priority=HIGHEST, name=FIREBASE_CRASHLYTICS_REPORT for destination cct
D/TransportRuntime.JobInfoScheduler(24514): Scheduling upload for context TransportContext(cct, HIGHEST, MSRodHRwczovL2NyYXNobHl0aWNzcmVwb3J0cy1wYS5nb29nbGVhcGlzLmNvbS92MS9maXJlbG9nL2xlZ2FjeS9iYXRjaGxvZ1xBSXphU3lCcnBTWVQ0RkZMMDlyZUhKaTZIOUZZZGVpU25VVE92Mk0=) with jobId=1526412857 in 1000ms(Backend next call timestamp 1611946367780). Attempt 1
I/Process (24514): Sending signal. PID: 24514 SIG: 9
Lost connection to device.


Keep the good work!
Thanks

**Comment by Jeasmine** [comment]
> Only tested on Android. I just updated the package on my existing project.
> 
> I want to test this version but the app terminates when it receives a new push notification in foreground. When the app is closed, the notification pop-up appears but the setNotificationOpenedHandler is not triggered.
> 
> **Foreground Error:**
> 
> > OneSignal(24514): OneSignalRestClient: After con.getResponseCode to: https://api.onesignal.com/players/5a56a8a3-bf4b-4ed3-aec9-bec223662f4c/on_session
> > D/OneSignal(24514): OneSignalRestClient: Failed request to: https://api.onesignal.com/players/5a56a8a3-bf4b-4ed3-aec9-bec223662f4c/on_session
> > W/OneSignal(24514): OneSignalRestClient: POST RECEIVED JSON: {"errors":["["Identifier invalid format."]"]}
> > W/OneSignal(24514): Failed last request. statusCode: 400
> > W/OneSignal(24514): response: {"errors":["["Identifier invalid format."]"]}
> > V/OneSignal(24514): initWithContext called with: android.app.ReceiverRestrictedContext@52807e1
> > V/OneSignal(24514): Starting OneSignal initialization!
> > V/OneSignal(24514): No class found, not setting up OSRemoteNotificationReceivedHandler
> > D/OneSignal(24514): OneSignal SDK initialization already completed.
> > D/OneSignal(24514): OSNotificationWorkManager enqueueing notification work with notificationId: aec40e9a-c0c4-4908-b8e6-44f0b3809be7 and jsonPayload: {"google.delivered_priority":"normal","google.sent_time":1611946374791,"google.ttl":259200,"google.original_priority":"normal","custom":"{"a":{"addition data":"1234","actionButtons":[],"actionId":"**DEFAULT**"},"i":"aec40e9a-c0c4-4908-b8e6-44f0b3809be7"}","pri":"5","vis":"1","from":"214972727859","alert":"asdasd","title":"Test Slack Alert","google.message_id":"0:1611946374815968%f54ecb25f9fd7ecd","google.c.sender.id":"214972727859"}
> > D/OneSignal(24514): NotificationWorker running doWork with data: androidx.work.Data@d0579111
> > D/OneSignal(24514): Running destroyTimeout with runnable: com.onesignal.OSNotificationReceivedEvent$1@5f80f8c
> > D/OneSignal(24514): Running startTimeout with timeout: 25000 and runnable: com.onesignal.OSNotificationReceivedEvent$1@5f80f8c
> > W/OneSignal(24514): remoteNotificationReceivedHandler not setup, displaying normal OneSignal notification
> > D/OneSignal(24514): Running destroyTimeout with runnable: com.onesignal.OSNotificationReceivedEvent$1@5f80f8c
> > D/OneSignal(24514): Starting processJobForDisplay opened: false fromBackgroundLogic: true
> > I/OneSignal(24514): Fire notificationWillShowInForegroundHandler
> > D/OneSignal(24514): Running destroyTimeout with runnable: com.onesignal.OSNotificationReceivedEvent$1@1a376d5
> > D/OneSignal(24514): Running startTimeout with timeout: 25000 and runnable: com.onesignal.OSNotificationReceivedEvent$1@1a376d5
> > D/onesignal(24514): Created json raw payload: {vis=1, google.original_priority=normal, alert=asdasd, google.sent_time=1611946374791, pri=5, google.delivered_priority=normal, custom={"a":{"addition data":"1234","actionButtons":[],"actionId":"**DEFAULT**"},"i":"aec40e9a-c0c4-4908-b8e6-44f0b3809be7"}, google.c.sender.id=214972727859, google.ttl=259200, from=214972727859, title=Test Slack Alert, google.message_id=0:1611946374815968%f54ecb25f9fd7ecd}
> > D/AndroidRuntime(24514): Shutting down VM
> > E/AndroidRuntime(24514): FATAL EXCEPTION: main
> > E/AndroidRuntime(24514): Process: com.domingosnunes.scorenoobapp, PID: 24514
> > E/AndroidRuntime(24514): java.lang.IllegalArgumentException: Unsupported value: {"addition data":"1234"}
> > E/AndroidRuntime(24514): 	at io.flutter.plugin.common.StandardMessageCodec.writeValue(StandardMessageCodec.java:278)
> > E/AndroidRuntime(24514): 	at io.flutter.plugin.common.StandardMessageCodec.writeValue(StandardMessageCodec.java:275)
> > E/AndroidRuntime(24514): 	at io.flutter.plugin.common.StandardMethodCodec.encodeMethodCall(StandardMethodCodec.java:38)
> > E/AndroidRuntime(24514): 	at io.flutter.plugin.common.MethodChannel.invokeMethod(MethodChannel.java:99)
> > E/AndroidRuntime(24514): 	at io.flutter.plugin.common.MethodChannel.invokeMethod(MethodChannel.java:83)
> > E/AndroidRuntime(24514): 	at com.onesignal.flutter.FlutterRegistrarResponder$4.run(FlutterRegistrarResponder.java:73)
> > E/AndroidRuntime(24514): 	at android.os.Handler.handleCallback(Handler.java:883)
> > E/AndroidRuntime(24514): 	at android.os.Handler.dispatchMessage(Handler.java:100)
> > E/AndroidRuntime(24514): 	at android.os.Looper.loop(Looper.java:214)
> > E/AndroidRuntime(24514): 	at android.app.ActivityThread.main(ActivityThread.java:7682)
> > E/AndroidRuntime(24514): 	at java.lang.reflect.Method.invoke(Native Method)
> > E/AndroidRuntime(24514): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
> > E/AndroidRuntime(24514): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
> > I/WM-WorkerWrapper(24514): Worker result SUCCESS for Work [ id=517e0eef-1a39-4340-9361-03eb804bf6ea, tags={ com.onesignal.OSNotificationWorkManager$NotificationWorker } ]
> > D/TransportRuntime.SQLiteEventStore(24514): Storing event with priority=HIGHEST, name=FIREBASE_CRASHLYTICS_REPORT for destination cct
> > D/TransportRuntime.JobInfoScheduler(24514): Scheduling upload for context TransportContext(cct, HIGHEST, MSRodHRwczovL2NyYXNobHl0aWNzcmVwb3J0cy1wYS5nb29nbGVhcGlzLmNvbS92MS9maXJlbG9nL2xlZ2FjeS9iYXRjaGxvZ1xBSXphU3lCcnBTWVQ0RkZMMDlyZUhKaTZIOUZZZGVpU25VVE92Mk0=) with jobId=1526412857 in 1000ms(Backend next call timestamp 1611946367780). Attempt 1
> > I/Process (24514): Sending signal. PID: 24514 SIG: 9
> > Lost connection to device.
> 
> Keep the good work!
> Thanks

Thanks for reporting this! will take a look

File: test/mock_channel.dart [hunk 2]
```diff
@@ -26,7 +26,7 @@ class OneSignalMockChannelController {
   Future<dynamic> _handleMethod(MethodCall call) async {
     print("Mock method called: ${call.method}");
     switch (call.method) {
-      case "OneSignal#init":
+      case "OneSignal#setAppId":
         this.state.initialize(call.arguments);
```
Comment (jkasten2) [inline test/mock_channel.dart:30]: Should rename this to `setAppId` as it seems that is all `initialize` does now.

File: ios/onesignal_flutter.podspec [hunk 3]
```diff
@@ -3,7 +3,7 @@
 #
 Pod::Spec.new do |s|
   s.name             = 'onesignal_flutter'
-  s.version          = '2.6.2'
+  s.version          = '3.0.0-beta2'
```
Comment (jkasten2) [inline ios/onesignal_flutter.podspec:6]: Should be 3.0.0-beta1

File: ios/onesignal_flutter.podspec [hunk 4]
```diff
@@ -13,7 +13,7 @@ Pod::Spec.new do |s|
   s.source_files = 'Classes/**/*'
   s.public_header_files = 'Classes/**/*.h'
   s.dependency 'Flutter'
-  s.dependency 'OneSignal', '2.16.1'
+  s.dependency 'OneSignal', '3.2.0'
   s.ios.deployment_target = '8.0'
```
Comment (jkasten2) [inline ios/onesignal_flutter.podspec:17]: Need to be upgraded to 9.0

**Review by jkasten2 — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/onesignal/flutter/OneSignalSerializer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jkasten2) [reply android/src/main/java/com/onesignal/flutter/OneSignalSerializer.java:64]: @Jeasmine @gonzalonarbaiz This still needs to be addressed

**Comment by jkasten2** [comment]
Also PR needs a rebase with master

File: android/build.gradle [hunk 5]
```diff
@@ -1,5 +1,5 @@
 group 'com.onesignal.flutter'
-version '2.6.2'
+version '3.0.0-beta2'
```
[the remark this answers is not in the document]
Comment (Jeasmine) [reply android/build.gradle:2]: oh my bad 🤦‍♀️

**Comment by Jeasmine** [comment]
> Also PR needs a rebase with master

I Will do after merging https://github.com/OneSignal/OneSignal-Flutter-SDK/pull/347 if that is ok

File: example/pubspec.yaml [hunk 6]
```diff
@@ -1,18 +1,20 @@
...
-  cupertino_icons: ^0.1.2
+  cupertino_icons: ^1.0.0
 
 dev_dependencies:
   flutter_test:
     sdk: flutter
 
-  onesignal_flutter: ^2.2.0
+  onesignal_flutter:
+    path: ^3.0.0-beta2
```
[the remark this answers is not in the document]
Comment (Jeasmine) [reply example/pubspec.yaml]: Addressed!

File: android/src/main/java/com/onesignal/flutter/OneSignalSerializer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jeasmine) [reply android/src/main/java/com/onesignal/flutter/OneSignalSerializer.java:64]: Actually, I checked, and this is only used for internal JSON parsing. It never arrives as "subscribed" to the user. Do you still want keys that are used internally to be named with the prefix "is"?

[the remark this answers is not in the document]
Comment (jkasten2) [reply android/src/main/java/com/onesignal/flutter/OneSignalSerializer.java:64]: ah ok, it is fine for this PR. We can clean it up for beta2 if it is just internal

**Comment by MeAtPros** [comment]
@Jeasmine Any update?

**Comment by MeAtPros** [comment]
@Jeasmine When will we get this update? Unfortunately, it blocks us from releasing a patch to our mobile application to which we added appropriate support to the latest version of flutter. Push notifications still do not trigger the appropriate callback after tapping on it.

**Comment by rickydj** [comment]
When the app executes the setNotificationOpenedHandler handler, it crashes with this stack;

Launching lib\main.dart on AOSP on IA Emulator in debug mode...
√ Built build\app\outputs\flutter-apk\app-debug.apk.
Connecting to VM Service at ws://127.0.0.1:55168/oOnyr4WWz-8=/ws
D/EGL_emulation( 9652): eglMakeCurrent: 0xcf809720: ver 3 0 (tinfo 0xcebef3d0)
D/eglCodecCommon( 9652): setVertexArrayObject: set vao to 0 (0) 1 0
I/tinel.teleappv( 9652): Waiting for a blocking GC ProfileSaver
I/tinel.teleappv( 9652): Background concurrent copying GC freed 21170(6MB) AllocSpace objects, 7(140KB) LOS objects, 49% free, 1539KB/3MB, paused 5.476ms total 58.407ms
I/tinel.teleappv( 9652): WaitForGcToComplete blocked ProfileSaver on ProfileSaver for 25.707ms
�[38;5;244m[GETX] �[39;49mInstance "GetMaterialController" has been created
�[38;5;244m[GETX] �[39;49mInstance "GetMaterialController" has been initialized
W/ComponentDiscovery( 9652): Class com.google.firebase.dynamicloading.DynamicLoadingRegistrar is not an found.
I/FirebaseApp( 9652): Device unlocked: initializing all Firebase APIs for app ONESIGNAL_SDK_FCM_APP_NAME
W/OneSignal( 9652): remoteNotificationReceivedHandler not setup, displaying normal OneSignal notification
D/onesignal( 9652): Created json raw payload: {vis=1, google.original_priority=normal, google.sent_time=1618523704868, pri=5, licon=https://img.onesignal.com/t/c183a670-3385-411d-9a81-5bfaccb64667.png, google.delivered_priority=normal, custom={"a":{"groupMessage":"registro"},"ti":"c04e4a65-3d08-4d53-8255-e5df45c2a3ec","i":"d145f360-75aa-4a85-8192-acbc6ae81e03","tn":"Registro"}, sound=id_com, bgn=1, google.c.sender.id=740258068655, oth_chnl=, title=Registro APP, google.message_id=0:1618523704879847%448e6de1f9fd7ecd, alert=Codigo de registro teleapp 441504, google.ttl=259200, from=740258068655}
D/AndroidRuntime( 9652): Shutting down VM
E/AndroidRuntime( 9652): FATAL EXCEPTION: main
E/AndroidRuntime( 9652): Process: com.telesentinel.teleappv3, PID: 9652
E/AndroidRuntime( 9652): java.lang.IllegalArgumentException: Unsupported value: {"groupMessage":"registro"}
E/AndroidRuntime( 9652): at io.flutter.plugin.common.StandardMessageCodec.writeValue(StandardMessageCodec.java:276)
E/AndroidRuntime( 9652): at io.flutter.plugin.common.StandardMessageCodec.writeValue(StandardMessageCodec.java:273)
E/AndroidRuntime( 9652): at io.flutter.plugin.common.StandardMethodCodec.encodeMethodCall(StandardMethodCodec.java:38)
E/AndroidRuntime( 9652): at io.flutter.plugin.common.MethodChannel.invokeMethod(MethodChannel.java:99)
E/AndroidRuntime( 9652): at io.flutter.plugin.common.MethodChannel.invokeMethod(MethodChannel.java:83)
E/AndroidRuntime( 9652): at com.onesignal.flutter.FlutterRegistrarResponder$4.run(FlutterRegistrarResponder.java:73)
E/AndroidRuntime( 9652): at android.os.Handler.handleCallback(Handler.java:873)
E/AndroidRuntime( 9652): at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime( 9652): at android.os.Looper.loop(Looper.java:193)
E/AndroidRuntime( 9652): at android.app.ActivityThread.main(ActivityThread.java:6669)
E/AndroidRuntime( 9652): at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 9652): at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
E/AndroidRuntime( 9652): at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
W/tinel.teleappv( 9652): Unknown chunk type '200'.
I/Process ( 9652): Sending signal. PID: 9652 SIG: 9
Lost connection to device.
Exited (sigterm)

**Comment by dgilperez** [comment]
@Jeasmine can you share any updates? Any ETA or progress?

We're testing v3 in production and it's not the best right now, we're seeing some issues already reported.

**Comment by Jeasmine** [comment]
> When the app executes the setNotificationOpenedHandler handler, it crashes with this stack;
> 
> Launching lib\main.dart on AOSP on IA Emulator in debug mode...
> √ Built build\app\outputs\flutter-apk\app-debug.apk.
> Connecting to VM Service at ws://127.0.0.1:55168/oOnyr4WWz-8=/ws
> D/EGL_emulation( 9652): eglMakeCurrent: 0xcf809720: ver 3 0 (tinfo 0xcebef3d0)
> D/eglCodecCommon( 9652): setVertexArrayObject: set vao to 0 (0) 1 0
> I/tinel.teleappv( 9652): Waiting for a blocking GC ProfileSaver
> I/tinel.teleappv( 9652): Background concurrent copying GC freed 21170(6MB) AllocSpace objects, 7(140KB) LOS objects, 49% free, 1539KB/3MB, paused 5.476ms total 58.407ms
> I/tinel.teleappv( 9652): WaitForGcToComplete blocked ProfileSaver on ProfileSaver for 25.707ms
> �[38;5;244m[GETX] �[39;49mInstance "GetMaterialController" has been created
> �[38;5;244m[GETX] �[39;49mInstance "GetMaterialController" has been initialized
> W/ComponentDiscovery( 9652): Class com.google.firebase.dynamicloading.DynamicLoadingRegistrar is not an found.
> I/FirebaseApp( 9652): Device unlocked: initializing all Firebase APIs for app ONESIGNAL_SDK_FCM_APP_NAME
> W/OneSignal( 9652): remoteNotificationReceivedHandler not setup, displaying normal OneSignal notification
> D/onesignal( 9652): Created json raw payload: {vis=1, google.original_priority=normal, google.sent_time=1618523704868, pri=5, licon=https://img.onesignal.com/t/c183a670-3385-411d-9a81-5bfaccb64667.png, google.delivered_priority=normal, custom={"a":{"groupMessage":"registro"},"ti":"c04e4a65-3d08-4d53-8255-e5df45c2a3ec","i":"d145f360-75aa-4a85-8192-acbc6ae81e03","tn":"Registro"}, sound=id_com, bgn=1, google.c.sender.id=740258068655, oth_chnl=, title=Registro APP, google.message_id=0:1618523704879847%448e6de1f9fd7ecd, alert=Codigo de registro teleapp 441504, google.ttl=259200, from=740258068655}
> D/AndroidRuntime( 9652): Shutting down VM
> E/AndroidRuntime( 9652): FATAL EXCEPTION: main
> E/AndroidRuntime( 9652): Process: com.telesentinel.teleappv3, PID: 9652
> E/AndroidRuntime( 9652): java.lang.IllegalArgumentException: Unsupported value: {"groupMessage":"registro"}
> E/AndroidRuntime( 9652): at io.flutter.plugin.common.StandardMessageCodec.writeValue(StandardMessageCodec.java:276)
> E/AndroidRuntime( 9652): at io.flutter.plugin.common.StandardMessageCodec.writeValue(StandardMessageCodec.java:273)
> E/AndroidRuntime( 9652): at io.flutter.plugin.common.StandardMethodCodec.encodeMethodCall(StandardMethodCodec.java:38)
> E/AndroidRuntime( 9652): at io.flutter.plugin.common.MethodChannel.invokeMethod(MethodChannel.java:99)
> E/AndroidRuntime( 9652): at io.flutter.plugin.common.MethodChannel.invokeMethod(MethodChannel.java:83)
> E/AndroidRuntime( 9652): at com.onesignal.flutter.FlutterRegistrarResponder$4.run(FlutterRegistrarResponder.java:73)
> E/AndroidRuntime( 9652): at android.os.Handler.handleCallback(Handler.java:873)
> E/AndroidRuntime( 9652): at android.os.Handler.dispatchMessage(Handler.java:99)
> E/AndroidRuntime( 9652): at android.os.Looper.loop(Looper.java:193)
> E/AndroidRuntime( 9652): at android.app.ActivityThread.main(ActivityThread.java:6669)
> E/AndroidRuntime( 9652): at java.lang.reflect.Method.invoke(Native Method)
> E/AndroidRuntime( 9652): at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
> E/AndroidRuntime( 9652): at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
> W/tinel.teleappv( 9652): Unknown chunk type '200'.
> I/Process ( 9652): Sending signal. PID: 9652 SIG: 9
> Lost connection to device.
> Exited (sigterm)

Hey @rickydj, thanks for the feedback. Were you trying to send additional data under Android notification? I found some problems in that scenario.

**Comment by rickydj** [comment]
Hi, today i solved the problem with the wrong value, just deletin the extra fields in the onesignal template, but now the system shows this:

I/WM-WorkerWrapper(12733): Worker result FAILURE for Work [ id=564a3c2e-9128-47ab-87c8-2b602e9ce600, tags={ com.onesignal.OSNotificationRestoreWorkManager$NotificationRestoreWorker } ]

and i cant receive those test notifications sent to my app

**Comment by Jeasmine** [comment]
> Hi, today i solved the problem with the wrong value, just deletin the extra fields in the onesignal template, but now the system shows this:
> 
> I/WM-WorkerWrapper(12733): Worker result FAILURE for Work [ id=564a3c2e-9128-47ab-87c8-2b602e9ce600, tags={ com.onesignal.OSNotificationRestoreWorkManager$NotificationRestoreWorker } ]
> 
> and i cant receive those test notifications sent to my app

Would you kindly provide me with your notification config? A screenshot of the config you are using on the dashboard will be enough to try the same config. Thanks!

File: android/build.gradle [hunk 7]
```diff
@@ -1,5 +1,5 @@
 group 'com.onesignal.flutter'
-version '2.6.3'
+version '3.0.0-beta4'
```
Comment (jkasten2) [inline android/build.gradle:2]: Remove -beta before merging into `main`

File: ios/onesignal_flutter.podspec [hunk 8]
```diff
@@ -3,7 +3,7 @@
 #
 Pod::Spec.new do |s|
   s.name             = 'onesignal_flutter'
-  s.version          = '2.6.4'
+  s.version          = '3.0.0-beta4'
```
Comment (jkasten2) [inline ios/onesignal_flutter.podspec:6]: Remove -beta before merging into `main`

File: example/pubspec.yaml [hunk 9]
```diff
@@ -1,18 +1,20 @@
...
-  cupertino_icons: ^0.1.2
+  cupertino_icons: ^1.0.0
 
 dev_dependencies:
   flutter_test:
     sdk: flutter
 
-  onesignal_flutter: ^2.2.0
+  onesignal_flutter:
+    path: ^3.0.0-beta4
```
Comment (jkasten2) [inline example/pubspec.yaml:17]: Remove -beta before merging into main

File: pubspec.yaml [hunk 10]
```diff
@@ -1,6 +1,6 @@
 name: onesignal_flutter
 description: OneSignal is a free push notification service for mobile apps. This plugin makes it easy to integrate your flutter app with OneSignal
-version: 2.6.4
+version: 3.0.0-beta4
```
Comment (jkasten2) [inline pubspec.yaml:3]: Remove -beta before merging into main

**Review by jkasten2 — CHANGES_REQUESTED** [review]

**Review by jkasten2 — APPROVED** [review]

**Pull request merged**